src/liballoc/macros.rs

```diff
@@ -39,9 +39,10 @@ macro_rules! vec {
     ($elem:expr; $n:expr) => (
         $crate::vec::from_elem($elem, $n)
     );
-    ($($x:expr),*) => (
-        <[_]>::into_vec(box [$($x),*])
-    );
+    ($($x:expr),*) => ({
+        let tmp = [$($x),*];
```

Comment:
Add a comment explaining why we do this...

Also add tests to ensure that the behavior won't regress.